From 6a649f540a217bb05130866e21270abdd94e7da1 Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Wed, 9 Dec 2015 20:03:01 +0800
Subject: [PATCH 1161/1565] drm/amd/dal: find proper connector in reset case

Find connector on reset properly, and allow reset mode occur for
MST connector.

It also remove warning messages seen previously

Change-Id: I6242ec305c7f4c81ffaa8b0c7280c512a6da66a3
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland<harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index 7023d2f..8753b28 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -2114,6 +2114,16 @@ int amdgpu_dm_atomic_commit(
 				aconnector = to_amdgpu_connector(connector);
 				break;
 			}
+
+			/*
+			 * this is the case when reset occur, connector is
+			 * removed from new crtc state. We need to update
+			 * connector state anyway. Access it from old_con_state
+			 */
+			if (old_con_state->crtc == crtc) {
+				aconnector = to_amdgpu_connector(connector);
+				break;
+			}
 		}
 
 		/* handles headless hotplug case, updating new_state and
-- 
1.9.1

